flake.nix

```diff
@@ -221,19 +221,26 @@
           cyclone = binDerivation {pkgName = "cyclone";};
 
           # autoPatchingElf and stripping will break deno compile'd binaries.
-          # They also have some very specific requirements about which glibc
-          # version to use and this conflicts what cyclone needs. These
-          # dependencies are added to the firecracker rootfs in rootfs_build.sh
-          # as it requires linking inside of the rootfs directly. Note that
-          # LD_LIBRAY_PATH is unset when using siExec to ensure the binaries our
-          # binaries run don't inherit it.
+          # We need to make sure we know where glibc and friends exist and since
+          # we can't patchelf, we need to ensure we drop the dynmic linker in a
+          # known place. Note that LD_LIBRAY_PATH is unset when using siExec to
+          # ensure the binaries our binaries run don't inherit it.
           lang-js = binDerivation {
             pkgName = "lang-js";
             dontAutoPatchELF = true;
             dontStrip = true;
             extraInstallPhase = ''
+              # since we can't patchelf, we need to ensure the dynamic linker
+              # is where we expect it. This gets droppped into the rootfs as
+              # /lib64/ld-linux-x86-64.so.2 -> /nix/store/*/ld-linux-x86-64.20.2
```

Comment:
Everything's built "hermetically" until it turns out that it doesn't portable run, eh? Pragmatism to the rescue, I love it